packages/firestore/src/model/collections.ts

```diff
@@ -37,6 +37,20 @@ export function maybeDocumentMap(): MaybeDocumentMap {
   return EMPTY_MAYBE_DOCUMENT_MAP;
 }
 
+export type NullableMaybeDocumentMap = SortedMap<
+  DocumentKey,
+  MaybeDocument | null
+>;
+
+export function nullableMaybeDocumentMap(): NullableMaybeDocumentMap {
+  return maybeDocumentMap();
+}
+
+export type DocumentSizeEntries = {
```

Comment:
I am nervous about this type. I know currently that the keys are kept in sync, but there is nothing here that guarantees it.